### PR TITLE
Fix the dark theme text color

### PR DESCRIPTION
**Pull request by GretaD**
Fixes https://github.com/nextcloud/mail/issues/1742

### CODE REVIEW

**Comment by mwalbeck** [comment]
I have tried out the fix and there is a small issue with the text colour of "#mail-message-has-blocked-content". "--color-primary-text" is set to white for both the dark and light themes which results in light text on a light background.

Screenshot_20210209_162832

Here it is with a dark text colour

Screenshot_20210209_162855

Other than that, it works as intended.

**Comment by GretaD** [comment]
@mwalbeck thank you for testing. But im not sure if i understood what do you mean. On both themes the "block contect string" it is readable. Do you mean it should be lighter?

**Comment by mwalbeck** [comment]
For me the text colour of the string "The images have been blocked to protect your privacy." is (```#ffffff```) and is therefore not visible at all.

I did a bit of digging in the server code and I may have found out why. The CSS variable ```--color-primary-text``` has a default value of ```#ffffff``` but it is dependent on the colour value set in the Theming app. If a colour that has a good contrast with white is chosen, like the default blue colour, then ```--color-primary-text``` is set to ```#ffffff```. But if a colour that has a bad contrast with white is chosen, for example green, then ```--color-primary-text``` is set to ```#000000```.

So if I set the colour in the Theming app to something like ```#5FC917``` then this fix works as intended and the string mentioning the blocked images is shown with text colour: ```#000000```. But with the default blue colour the string is shown with text colour: ```#ffffff```.

I think a hardcoded colour value may be best here as we want it to always be dark since it will be shown on a light background no matter what theme is being used.

**Comment by GretaD** [comment]
> For me the text colour of the string "The images have been blocked to protect your privacy." is (`#ffffff`) and is therefore not visible at all.
> 
> I did a bit of digging in the server code and I may have found out why. The CSS variable `--color-primary-text` has a default value of `#ffffff` but it is dependent on the colour value set in the Theming app. If a colour that has a good contrast with white is chosen, like the default blue colour, then `--color-primary-text` is set to `#ffffff`. But if a colour that has a bad contrast with white is chosen, for example green, then `--color-primary-text` is set to `#000000`.
> 
> So if I set the colour in the Theming app to something like `#5FC917` then this fix works as intended and the string mentioning the blocked images is shown with text colour: `#000000`. But with the default blue colour the string is shown with text colour: `#ffffff`.
> 
> I think a hardcoded colour value may be best here as we want it to always be dark since it will be shown on a light background no matter what theme is being used.

ok, i dont have any message to test your "edge case" but you are right on your findings. With `--color-primary-text` all my html messages with blocked content were shown ok in both themes. But i changed it as you suggested. Please have a look.

**Comment by mwalbeck** [comment]
The latest commit works as intended on my end. Cheers!

File: src/components/MessageHTMLBody.vue [hunk 1]
```diff
@@ -67,7 +67,7 @@ export default {
 				// send css vars to client page
 				this.$refs.iframe.iFrameResizer.sendMessage({
 					cssVars: {
-						...getCssVar('--color-main-text'),
+						...getCssVar('--color-main-background'),
```
Comment (st3iny) [inline src/components/MessageHTMLBody.vue:70]: Just sending another css variable to the client page won't change anything on its own. If we scrap the injection of the text color and just force a white background, this whole block can be removed (including the `getCssVar()` method).

See the injected css for reference (`css/html-response.css`):
```css
* {
	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Cantarell, Ubuntu, 'Helvetica Neue', Arial, 'Noto Color Emoji', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
}

body {
	color: var(--color-main-text);
}
```

The whole body block can be removed aswell from the css file.

**Review by st3iny — APPROVED** [review]
Tested on my local instance and mails are displayed correctly when using a dark theme. Forcing a white background works a lot better than my former injection hack.

**Comment by ChristophWurst** [comment]
Let's see if the backport works, then we can do this for 1.8.x, otherwise we have this only in 1.9
